### ISSUE TITLE
Crash with BiblioCraft 1.10.5

### ISSUE THREAD

**Issue by xJon** [Nuchaz/BiblioCraft#305]
I got this report from a user, also I see some things related to CoFH.
What causes that? Is it something with the latest update?
http://paste.ubuntu.com/11960802/
Thanks!

**Comment by Nuchaz** [comment]
I don't believe this is related to the 1.10.5 update. This is that one odd issue that comes up from time to time where the block id mappings for the world gets confused so you end up with blocks on top of tile entities for other blocks and when the block tried to access its tile entity and finds something else, it usually crashes. I tried to put in some extra safeguards in 1.10.5 to reduce the actual crashes when it happens, but it wouldn't fix the world itself. I think this issue can happen with any mods and isn't necessarily specific to BiblioCraft. BiblioCraft just has a lot of blocks with tileentities so this easily effect BiblioCraft. 

It looks like CoFH is trying to get the collision boxes for all the blocks and since some of the BiblioCraft blocks are on the wrong tile entries it crashes. Specifically, there is a Bookcase block on top of a potion shelf tileentity in this case. The BiblioCraft blocks have to read the angle of the block from the tile entity to figure out the collision block, so that it was it is crashing when CoFH tried to get the collision blocks.

I went ahead and put in safeguards against the crash for the next release. That was one of the areas I tried to safeguard across all my blocks, but I guess I had missed the bookcase.
